ts/features/fci/screens/valid/FciDocumentsScreen.tsx

```diff
@@ -23,16 +38,191 @@ const styles = StyleSheet.create({
   }
 });
 
+export type FciDocumentsScreenNavigationParams = Readonly<{
+  attrs: SignatureField["attrs"];
+  currentDoc: number;
+}>;
+
+type SignatureFieldAttrType =
+  | ExistingSignatureFieldAttrs
+  | SignatureFieldToBeCreatedAttrs;
+
+const hasUniqueName = (
+  f: SignatureFieldAttrType
+): f is ExistingSignatureFieldAttrs =>
+  (f as ExistingSignatureFieldAttrs).unique_name !== undefined;
+
 const FciDocumentsScreen = () => {
   const pdfRef = React.useRef<Pdf>(null);
   const [totalPages, setTotalPages] = React.useState(0);
   const [currentPage, setCurrentPage] = React.useState(0);
   const [currentDoc, setCurrentDoc] = React.useState(0);
+  const [signaturePage, setSignaturePage] = React.useState(0);
+  const [pdfString, setPdfString] = React.useState<string>("");
   const documents = useSelector(fciSignatureDetailDocumentsSelector);
   const navigation = useNavigation();
+  const route = useRoute<RouteProp<FciParamsList, "FCI_DOCUMENTS">>();
+  const attrs = route.params.attrs as
+    | ExistingSignatureFieldAttrs
+    | SignatureFieldToBeCreatedAttrs;
+  const cDoc = route.params.currentDoc;
+  const documentSignaturesSelector = useSelector(fciDocumentSignaturesSelector);
+  const dispatch = useIODispatch();
+
+  React.useEffect(() => {
+    // if the user hasn't checked any signauture field,
+    // we need to initialize the documentSignatures state
+    if (RA.isEmpty(documentSignaturesSelector)) {
+      pipe(
+        documents,
+        RA.map(d => {
+          const docSignature = {
+            document_id: d.id,
+            signature: "",
+            signature_fields: []
+          } as DocumentSignature;
+          dispatch(fciUpdateDocumentSignaturesRequest(docSignature));
+        })
+      );
+    }
+  }, [dispatch, documentSignaturesSelector, documents]);
+
+  /**
+   * function to draw a rect over a signature field
+   * @param uniqueName the of the signature field
+   */
+  const drawRectangleOverSignatureFieldById = React.useCallback(
+    async (uniqueName: string) => {
+      // TODO: refactor this function to use fp-ts
+      const doc = documents[currentDoc];
+      const url = doc.url;
+
+      const existingPdfBytes = await ReactNativeBlobUtil.fetch("GET", url).then(
+        res => res.base64()
+      );
+
+      await PDFDocument.load(
+        `data:application/pdf;base64,${existingPdfBytes}`
+      ).then(res => {
+        // get the signature field by unique name
+        pipe(
```

Comment:
what if this throws?

Reply:
no react is displayed for the signature field.